src/views/FileExplorer.ts

```diff
@@ -381,38 +369,53 @@ export class FileSystemProvider implements vscode.TreeDataProvider<Entry>, vscod
 
     if (element) {
       const children = await this.readDirectory(element);
-      return children.map(([name, type]) => Object.assign(vscode.Uri.file(path.join(element.fsPath, name)), {type}));
+      return children.map(([name, type]) =>
+        Object.assign(vscode.Uri.file(path.join(element.fsPath, name)), {type, isContractFolder: false})
+      );
+    }
+
+    // Gets the workspace folders
+    const workspaces = await getAllTruffleWorkspaces();
+
+    // Checks if there are any workspaces
+    if (workspaces.length === 0) {
+      Output.outputLine(OutputLabel.truffleForVSCode, Constants.errorMessageStrings.TruffleConfigIsNotExist);
+      vscode.window.showInformationMessage(Constants.errorMessageStrings.TruffleConfigIsNotExist);
+
+      return [];
     }
 
-    const workspaceFolder = getWorkspaceFolder();
-    if (workspaceFolder) {
-      let children = await this.getSortedChildren(workspaceFolder.uri);
-
-      if (this._baseFolder) {
-        // we need to filter for this folder
-        const baseFolderUri = children?.filter((cVal) => {
-          return cVal[1] == vscode.FileType.Directory && cVal[0] === this._baseFolder;
-        });
-        // if we find it we change our children to be its.
-        if (baseFolderUri && baseFolderUri.length > 0) {
-          // just set this to our baseFolder.
-          children = children.filter((v) => this._baseFolder?.localeCompare(v[0]) == 0);
-          Output.outputLine(OutputLabel.truffleForVSCode, `Setting Base Folder to: ${this._baseFolder}`);
-        } else {
-          Output.outputLine(
-            OutputLabel.truffleForVSCode,
-            `no baseFolder: ${this._baseFolder} found in children of workspace: ${children}`
-          );
-          vscode.window.showInformationMessage(`No folder "${this._baseFolder}" found in workspace.`);
-          return [];
-        }
+    // In some cases we may have more than one truffle configuration file, so it's better to remove duplicates item from the array
+    const uniqueWorkspaces = Array.from(new Set(workspaces.map((workspace) => workspace.dirName))).map((dirName) => {
```

Comment:
Not sure if I understand this, why are we removing duplicates? Some boxes have more than one Truffle config file, so we need to display all of them. Otherwise, the _Contract Explorer_ and the _Deployments_ views display inconsistent data

<img width="278" alt="image" src="https://user-images.githubusercontent.com/4592980/199503349-96e9b4ca-7987-4537-80ab-cd5c7bf4df41.png">